src/main/java/tetris/Action.java

```diff
@@ -14,12 +14,19 @@ public String toString() {
         @Override
         public void attempt(AbstractTetromino tetromino, Grid grid) {
             tetromino.rotateRight();
+
             for (int i = 0; i < 4; i++) {
-                if (!grid.isFree(tetromino.get(i))) {
-                    tetromino.rotateLeft();
+                if (!grid.isFree(tetromino.leftCoor())) {
+                    tetromino.moveRight();
```

Comment:
Like how you're thinking, but there are some bordercases not covered by this solution. I managed to make a very illegal move: 
![image](https://cloud.githubusercontent.com/assets/10614357/19533938/18da8ae4-9642-11e6-9261-5c73b54f5789.png)